src/rt/test/perf/banking/pthread_banks.cc

```diff
@@ -0,0 +1,66 @@
+// Copyright Microsoft and Project Verona Contributors.
+// SPDX-License-Identifier: MIT
+#include "args.h"
+
+#include <algorithm>
+#include <cassert>
+#include <iostream>
+#include <mutex>
+#include <test/harness.h>
+#include <thread>
+#include <vector>
+
+namespace
+{
+  /// Protects the state of the accounts (note, in this benchmark there is no state)
+  std::mutex* accounts;
+
+  /// Protects the state of the logger (note, in this benchmark that is tx_count)
+  std::mutex logger;
+  
+  /// Something representing the loggers state.
+  uint64_t tx_count = 0;
+}
+
+void thread_main(size_t id)
+{
+  size_t a, b;
+  xoroshiro::p128r64 rand;
+  rand.set_state(id + 1);
+
+  for (size_t i = 0; i < (TRANSACTIONS / NUM_WORKERS); i++)
+  {
+    a = rand.next() % NUM_ACCOUNTS;
+    b = a;
+    while (b == a)
```

Comment:
If `NUM_ACCOUNTS == 1`, this will loop forever. :)

Reply:
This is checked here:
https://github.com/microsoft/verona/pull/572/files#diff-c36710122675c2f451aed2355723cde9536ee9f011ed0ffd7a9d138ac1d09b27R22-R26